### PR TITLE
La 314 downstream exceptions handling

### PR DESCRIPTION
**Pull request by Vadym-Lopatka**
[LA-314] downstream exceptions handling

[LA-314]: https://selina.atlassian.net/browse/LA-314?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **vadym-lopatka**
You can assign the PR to them by writing `/assign @vadym-lopatka` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-141** [here](http://ms-lending-api-pr141.selinadev.co.uk)

**Review by bpoon-selina — CHANGES_REQUESTED** [review]

File: src/main/java/com/selina/lending/api/errors/ExceptionTranslator.java [hunk 1]
```diff
@@ -89,6 +93,10 @@ public ProblemBuilder prepare(@NotNull final Throwable throwable, @NotNull final
         return buildProblem(status, throwable.getMessage(), throwable);
     }
 
+    private boolean isThisTypeShouldBeSuppressed(FeignException feignException) {
```
[the remark this answers is not in the document]
Comment (Vadym-Lopatka) [reply src/main/java/com/selina/lending/api/errors/ExceptionTranslator.java:96]: how about 404?

File: src/main/java/com/selina/lending/internal/repository/auth/AuthRepositoryImpl.java [hunk 2]
```diff
@@ -35,13 +37,20 @@ public AuthRepositoryImpl(AuthApi authApi) {
...
+        try {
+            var apiResponse = authApi.login(Map.of(
+                            "client_id", credentials.clientId(),
+                            "client_secret", credentials.clientSecret(),
+                            "grant_type", "client_credentials"
+                    )
+            );
+            return new TokenResponse(apiResponse.access_token(), apiResponse.expires_in());
+        } catch (FeignException feignException) {
+            if (feignException.status() == 400) {
```
[the remark this answers is not in the document]
Comment (Vadym-Lopatka) [reply src/main/java/com/selina/lending/internal/repository/auth/AuthRepositoryImpl.java:49]: yes, I wanted to propose it to you as the next, separate task.
I think smth like this should be our target:

{
"title": "Bad Request",
"status": 400,
"detail": "Invalid client credentials"
}

[the remark this answers is not in the document]
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/internal/repository/auth/AuthRepositoryImpl.java:49]: Can it be done in this task, or is it a more sizeable task to do?

[the remark this answers is not in the document]
Comment (Vadym-Lopatka) [reply src/main/java/com/selina/lending/internal/repository/auth/AuthRepositoryImpl.java:49]: ok, I will fix as a part of this

File: src/main/java/com/selina/lending/api/errors/ExceptionTranslator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/api/errors/ExceptionTranslator.java:96]: 404, should be returned to the client as:
{
    "title": "Not Found",
    "status": 404,
    "detail": "Application not found for xxx"
}
But i think at the moment, its passing the error straight up to the client from middleware

[the remark this answers is not in the document]
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/api/errors/ExceptionTranslator.java:96]: For example:
{
    "title": "Not Found",
    "status": 404,
    "detail": "{\"exception_type\":\"ApplicationByExternalIdException\",\"message\":\"Application with external id: xx does not exist.\"}"
}

[the remark this answers is not in the document]
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/api/errors/ExceptionTranslator.java:96]: Agreed that a new ticket will handle this 404 formating and all other httpstatus codes should be suppressed

File: src/main/java/com/selina/lending/internal/repository/auth/AuthRepositoryImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/internal/repository/auth/AuthRepositoryImpl.java:49]: Agreed this will be done in a new ticket along with the 404

**Pull request merged**